Source/AwsCommonRuntimeKit/http/HTTPClientConnection.swift

```diff
@@ -25,21 +40,27 @@ public class HTTPClientConnection {
         return aws_http_connection_is_open(rawValue)
     }
 
+    public var httpVersion: HTTPVersion {
+        HTTPVersion(rawValue: aws_http_connection_get_version(rawValue))
+    }
+
     /// Close the http connection
     public func close() {
         aws_http_connection_close(rawValue)
     }
 
     /// Creates a new http stream from the `HTTPRequestOptions` given.
+    /// The stream will send no data until HTTPStream.activate()
+    /// is called. Call activate() when you're ready for callbacks and events to fire.
     /// - Parameter requestOptions: An `HTTPRequestOptions` struct containing callbacks on
     /// the different events from the stream
     /// - Returns: An `HTTPStream` containing the `HTTPClientConnection`
     public func makeRequest(requestOptions: HTTPRequestOptions) throws -> HTTPStream {
         let httpStreamCallbackCore = HTTPStreamCallbackCore(requestOptions: requestOptions)
         do {
-            return try HTTPStream(httpConnection: self,
-                                  options: httpStreamCallbackCore.getRetainedHttpMakeRequestOptions(),
-                                  callbackData: httpStreamCallbackCore)
+            return try HTTP1Stream(httpConnection: self,
+                                   options: httpStreamCallbackCore.getRetainedHttpMakeRequestOptions(),
+                                   callbackData: httpStreamCallbackCore)
```

Comment:
debatable: we could move this to an `HTTP1ClientConnection` class

Reply:
We will need to create a protocol which have a makeRequest function so that we can call makeRequest on the `HTTPClientConnectionBase` which almost gives us the same solution as HTTPExchange previously. 

We can do it if there can be things that might only apply to HTTP1Connection but not HTTP2Connection given by ConnectionManager.